### PR TITLE
feat: repository function name as action type

### PR DESCRIPTION
**Pull request by soyuka**
It's a rather naive and maybe not 100% functional way of doing it but it's prettier:

20211117_18h16m37s_grim

Feel free to close this if it isn't worth it in your opinion.

### CODE REVIEW

**Comment by NetanelBasal** [comment]
Thanks for the PR. What about accepting a callback function that will run from the outside and will return the action type? That way, this code could be provided only for consumers that want it.

**Comment by soyuka** [comment]
Indeed this'd be even better, you mean a callback to the `devtools` function?

**Comment by NetanelBasal** [comment]
Yes

**Comment by soyuka** [comment]
Done :)

File: packages/devtools/src/lib/devtools.spec.ts [hunk 1]
```diff
@@ -83,7 +87,7 @@ describe('devtools', () => {
       }
     );
 
-    createEntitiesStore('foo');
```
Comment (NetanelBasal) [inline packages/devtools/src/lib/devtools.spec.ts:86]: What are the reasons for these changes?

File: packages/devtools/src/lib/devtools.spec.ts [hunk 2]
```diff
@@ -145,4 +149,52 @@ describe('devtools', () => {
...
+
+  Object.defineProperty(global, 'window', {
+    value: {
+      __REDUX_DEVTOOLS_EXTENSION__: {
+        connect: jest.fn().mockImplementation(() => instanceMock),
+      },
+    },
+  });
+
+  devTools({ getActionType: (_: any): string => 'Hello' });
```
Comment (NetanelBasal) [inline packages/devtools/src/lib/devtools.spec.ts:175]: Why do we need to pass the state?

File: packages/devtools/src/lib/devtools.spec.ts [hunk 3]
```diff
@@ -145,4 +149,52 @@ describe('devtools', () => {
...
+
```
Comment (NetanelBasal) [inline packages/devtools/src/lib/devtools.spec.ts:176]: Can you write the spec with your example code, please?

**Comment by NetanelBasal** [comment]
I closed it by mistake, please reopen it.

**Pull request closed without merging**